### PR TITLE
Support centralised environments in `uv venv`

### PR DESCRIPTION
**Pull request by EliteTK**
## Summary

With the centralised project environments preview feature enabled, `uv venv` invoked with no path and `UV_PROJECT_ENVIRONMENT` unset  will now use the environment cache bucket as the backing store for the environment. The key selection logic is the same as for centralised project environments. 

There's are some cases of note:

* With the feature enabled:
  * Unless you explicitly pass `--no-clear`, existing environments will be clobbered without prompting
  * If you pass `--allow-existing` this will avoid clearing the _centralised_ environment, before re-using it, but will still clobber local environments when creating the link. This might sound weird, but the alternatives are:
    * Disable the feature if `.venv` exists, but if `.venv` doesn't exist, instead operate on any "existing" environment in the cache (this gets more confusing the more you think about it)
    * Don't touch `.venv` if it exists, but otherwise operate on any "existing" environment in the cache
    * Complain if `.venv` exists but is not referencing the cache (somewhat sane sounding... except also somewhat heavy and still weird)
* With the feature disabled:
  * Whenever we would normally clear an environment, if it's a link to the cache, we just delete the link (we would normally clear the target of the link). I consider this not to be breaking because a `.venv` that is a link into the cache should never have been a thing prior to the feature.
  * The user is not prompted to remove links to the cache.

See also #18214.

## Test Plan

A range of new tests covering a variety of normal and corner cases.

### CODE REVIEW

File: crates/uv/src/commands/project/mod.rs [hunk 1]
```diff
@@ -1203,11 +1203,11 @@ pub(crate) enum LinkErrorReporting {
 }
 
 /// Point the workspace's `.venv` to the centralized environment.
```
Comment (zsol) [inline crates/uv/src/commands/project/mod.rs:1205]: this needs an update mentioning the return value

**Review by zsol — APPROVED** [review]
does `uv cache prune` remove orphaned envs after someone switches off the preview flag and recreates their `.venv`?

**Comment by EliteTK** [comment]
We have no way to detect an in-use `.venv` because it could be on a currently unmounted filesystem or another machine. We don't have a project database and symlinks don't increment anything similar to the hardlink count. We have resigned to the fact that we don't have a way to determine what "orphaned" means and we clear all cached environments always, regardless of the feature.

**Review by zanieb — APPROVED** [review]

File: crates/uv/src/commands/project/mod.rs (hunk 1 above)
Comment (zanieb) [reply crates/uv/src/commands/project/mod.rs:1205]: https://github.com/astral-sh/uv/pull/20010
